dsl_parser/functions.py

```diff
@@ -592,6 +547,103 @@ def _validate_ref(self, ref, ref_name):
                                        self.attribute_path))
 
 
+def _get_attribute_from_node_instance(ni, node, attribute_path, path):
+    value = _get_property_value(ni['node_id'],
+                                ni.get('runtime_properties'),
+                                attribute_path,
+                                path,
+                                raise_if_not_found=False)
+    if value is None:
+        # attribute not found in instance runtime properties
+        if value is None:
```

Comment:
eh?

Reply:
Whoops, fixing

Reply:
That's where was refactoring, must've typed one out then copied the other in, fixed.